app/(main)/league/[leagueId]/entry/[entryId]/week/Week.test.tsx

```diff
@@ -119,6 +124,100 @@ const updatedWeeklyPicks = {
   },
 };
 
+describe('Entry Name Editiing', () => {
+  beforeEach(() => {
+    mockUseAuthContext.isSignedIn = true;
+    (getCurrentUserEntries as jest.Mock).mockResolvedValue([
+      {
+        $id: '123',
+        name: 'Entry 1',
+        user: '123',
+        league: '123',
+        selectedTeams: [],
+        eliminated: false,
+      },
+    ]);
+  });
+
+  it('should display the edit button', async () => {
+    render(
+      <Week entry={entry} league={league} NFLTeams={NFLTeams} week={week} />,
+    );
+
+    await waitFor(() => {
+      const editEntryNameButton = screen.getByTestId('edit-entry-name-button');
+      expect(editEntryNameButton).toBeInTheDocument();
+    });
+  });
+
+  it('should switch to edit mode when the button is clicked', async () => {
+    render(
+      <Week entry={entry} league={league} NFLTeams={NFLTeams} week={week} />,
+    );
+
+    await waitFor(() => {
+      expect(screen.getByTestId('edit-entry-name-button')).toBeInTheDocument();
+    });
+    fireEvent.click(screen.getByTestId('edit-entry-name-button'));
+
+    const entryNameInput = screen.getByTestId('entry-name-input');
+    const cancelButton = screen.getByLabelText('Cancel Editing');
```

Comment:
Do not grab elements by text. If the text changes, this test will fail. Instead, add `data-testid` attribute. This is a unique ID that won't ever change. On top of this, when we add localization, things won't always be in English.